tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -937,31 +975,86 @@ def _generate_build_extra_metadata_for_tests(tests, bazel_rules):
     'deps("//src/proto/...")',
 ]
 
+# Step 1: run a bunch of "bazel query --output xml" queries to collect
+# the raw build metadata from the bazel build.
+# At the end of this step we will have a dictionary of bazel rules
+# that are interesting to us (libraries, binaries, etc.) along
+# with their most important metadata (sources, headers, dependencies)
```

Comment:
Can you show an example of what the resulting rule dict will look like?

Reply:
done.